operators/pkg/forge/containers.go

```diff
@@ -0,0 +1,328 @@
+		ContainerPort: port,
+		Protocol:      corev1.ProtocolTCP,
+	})
+}
+
+// AddEnvVariableToContainer appends an environment variable to the given container's env with the specified value.
+func AddEnvVariableToContainer(c *corev1.Container, name, value string) {
+	c.Env = append(c.Env, corev1.EnvVar{
+		Name:  name,
+		Value: value,
+	})
+}
+
+// AddEnvVariableFromResourcesToContainer appends an environment variable to the given container's env with a resource-referenced value.
+func AddEnvVariableFromResourcesToContainer(c *corev1.Container, name string, resName corev1.ResourceName) {
+	c.Env = append(c.Env, corev1.EnvVar{
+		Name: name,
+		ValueFrom: &corev1.EnvVarSource{
+			ResourceFieldRef: &corev1.ResourceFieldSelector{
+				ContainerName: c.Name,
+				Resource:      resName.String(),
+			},
+		},
+	})
+}
+
+// AddContainerVolumeMount appends a VolumeMount to the given container's volumeMounts.
+func AddContainerVolumeMount(c *corev1.Container, name, path string) {
+	c.VolumeMounts = append(c.VolumeMounts, corev1.VolumeMount{
+		Name:      name,
+		MountPath: path,
+	})
+}
+
+// AddContainerArg appends an argument to the given container's args in the format of --name=value.
+func AddContainerArg(c *corev1.Container, name, value string) {
+	c.Args = append(c.Args, fmt.Sprintf("--%s=%s", name, value))
+}
+
+// SetContainerReadinessTCPProbe sets the given container's ReadinessProbe with a TCPSocket handler.
+func SetContainerReadinessTCPProbe(c *corev1.Container, portName string) {
+	probe := ContainerProbe()
+	probe.Handler.TCPSocket = &corev1.TCPSocketAction{
+		Port: intstr.FromString(portName),
+	}
+	c.ReadinessProbe = probe
+}
+
+// SetContainerReadinessHTTPProbe sets the given container's ReadinessProbe with a HTTPGet handler.
+func SetContainerReadinessHTTPProbe(c *corev1.Container, portName, path string) {
+	probe := ContainerProbe()
+	probe.Handler.HTTPGet = &corev1.HTTPGetAction{
+		Port: intstr.FromString(portName),
+		Path: path,
+	}
+	c.ReadinessProbe = probe
+}
+
+// ContainerProbe forges a Probe with the following values:
+// InitialDelaySeconds: 10
+// PeriodSeconds:       2
+// SuccessThreshold:    2
+// FailureThreshold:    5
+// It sets no handler.
+func ContainerProbe() *corev1.Probe {
+	return &corev1.Probe{
+		InitialDelaySeconds: 10,
+		PeriodSeconds:       2,
+		SuccessThreshold:    2,
+		FailureThreshold:    5,
+		Handler:             corev1.Handler{},
+	}
+}
+
+// SetContainerResources sets the given container's Resources with the given values:
+// cpu values are converted to millis: 1.2 -> 1200m; ram sizes are in MiB.
+func SetContainerResources(
+	c *corev1.Container,
+	cpuRequests, cpuLimits float32,
+	memRequestsMi, memLimitsMi int,
+) {
+	c.Resources = corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    *resource.NewMilliQuantity(int64(cpuRequests*1e3), resource.DecimalSI),
+			corev1.ResourceMemory: *resource.NewQuantity(int64(memRequestsMi*(1<<20)), resource.BinarySI),
```

Comment:
Please, don't do binary operations. Use `resource.NewScaledQuantity` which is self explanatory (and be consistent, always using the DecimalSI).

Reply:
It just seemed weird to use decimal in case of memory since I basically always saw it writtein like 1024Mi and such stuff. I haven't used scaledquantity since it only creates values with decimalSI if i'm not wrong, but if it's actually preferred to use the decimal then I'll clearly switch to newScaledQuantity

Reply:
It just seemed weird to use decimal in case of memory since I basically always saw it writtein like 1024Mi and such stuff. I haven't used scaledquantity since it only creates values with decimalSI if i'm not wrong, but if it's actually preferred to use the decimal then I'll clearly switch to newScaledQuantity.

Reply:
Fair enough. I feel it could be probably fine to use the DecimalSI, but I understand your point. As you prefer